operator/metrics/metrics.go

```diff
@@ -79,5 +84,30 @@ func initializeMetrics(p params) {
 	p.Registry.MustRegister(metrics.ErrorsWarnings)
 	metrics.FlushLoggingMetrics()
 
-	p.Registry.AddServerRuntimeHooks()
+	// Initialize Prometheus server with default config
+	p.Registry.InitalizeServer()
+
+	p.JobGroup.Add(job.OneShot("operator-prometheus-server", func(ctx context.Context, _ cell.Health) error {
+		tlsEnabled := p.PrometheusTlsConfigPromise != nil
+		if tlsEnabled {
+			p.Logger.Info("Waiting for TLS certificates to become available")
+			certLoaderWatchedServerConfig, err := p.PrometheusTlsConfigPromise.Await(ctx)
+			if err != nil {
+				return err
+			}
+
+			// Config Prometheus server to run with TLS
+			p.Registry.ConfigureServerTLS(certLoaderWatchedServerConfig.ServerConfig(&tls.Config{
+				MinVersion: tls.VersionTLS13,
+			}))
+		}
+		go p.Registry.StartServer()
```

Comment:
This is already in its own job, so I don't think it needs to spawn another goroutine.

```suggestion
		p.Registry.StartServer()
```

Reply:
This is outdated, but i still follow your suggestion not to spawn another routine in OneShot.